src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/bulkscan/AttachScannedDocsCallbackHandlerTest.java

```diff
@@ -0,0 +1,181 @@
+import uk.gov.hmcts.cmc.ccd.domain.CCDScannedDocument;
+import uk.gov.hmcts.cmc.ccd.domain.CaseEvent;
+import uk.gov.hmcts.cmc.ccd.domain.defendant.CCDRespondent;
+import uk.gov.hmcts.cmc.ccd.domain.defendant.CCDResponseMethod;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.CallbackParams;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.CallbackType;
+import uk.gov.hmcts.cmc.claimstore.utils.CaseDetailsConverter;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.cmc.ccd.domain.defendant.CCDResponseMethod.OFFLINE;
+
+@ExtendWith(MockitoExtension.class)
+class AttachScannedDocsCallbackHandlerTest {
+
+    @Mock
+    private CaseDetailsConverter caseDetailsConverter;
+
+    private AttachScannedDocsCallbackHandler callbackHandler;
+
+    private CallbackParams callbackParams;
+
+    @BeforeEach
+    void setUp() {
+        callbackHandler = new AttachScannedDocsCallbackHandler(caseDetailsConverter);
+
+        CallbackRequest callbackRequest = CallbackRequest
+            .builder()
+            .caseDetails(CaseDetails.builder().data(Collections.EMPTY_MAP).build())
+            .eventId(CaseEvent.ATTACH_SCANNED_DOCS.getValue())
+            .build();
+
+        callbackParams = CallbackParams.builder()
+            .type(CallbackType.ABOUT_TO_SUBMIT)
+            .request(callbackRequest)
+            .build();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"N9a", "N9b", "N11", "N225", "N180"})
+    void shouldSetOfflineResponseMethodForResponseScannedDocuments(String documentCode) {
+        CCDCollectionElement<CCDScannedDocument> scannedDocument = CCDCollectionElement.<CCDScannedDocument>builder()
+            .value(CCDScannedDocument.builder().subtype(documentCode).build())
+            .build();
+        CCDCollectionElement<CCDRespondent> respondent = CCDCollectionElement.<CCDRespondent>builder()
+            .value(CCDRespondent.builder().build())
+            .build();
+
+        CCDCase ccdCase = CCDCase.builder()
+            .scannedDocuments(List.of(scannedDocument))
+            .respondents(List.of(respondent))
+            .build();
+
+        when(caseDetailsConverter.extractCCDCase(any())).thenReturn(ccdCase);
+
+        CCDCollectionElement<CCDRespondent> expectedRespondent = CCDCollectionElement.<CCDRespondent>builder()
+            .value(CCDRespondent.builder().responseMethod(OFFLINE).build())
+            .build();
+
+        CCDCase expectedCcdCase = ccdCase.toBuilder()
+            .respondents(List.of(expectedRespondent))
+            .build();
+
+        when(caseDetailsConverter.convertToMap(expectedCcdCase)).thenReturn(Map.of());
+
+        callbackHandler.handle(callbackParams);
+
```

Comment:
What are you actually verifying? Just that no exceptions are thrown?

Reply:
Added explicit verify